core/dbt/contracts/graph/nodes.py

```diff
@@ -241,10 +245,53 @@ class ParsedNode(NodeInfoMixin, ParsedNodeMandatory, SerializableType):
     unrendered_config: Dict[str, Any] = field(default_factory=dict)
     created_at: float = field(default_factory=lambda: time.time())
     config_call_dict: Dict[str, Any] = field(default_factory=dict)
+    constraints_enabled: bool = False
     relation_name: Optional[str] = None
     raw_code: str = ""
 
+    def constraints_validator(self):
```

Comment:
At first glance, this appears to be four types of errors (schema, materialization, language, and data type) in the same method. Can we break them out into separate methods? Perhaps this is a controller that provides a single interface, but there's a lot going on here.

Reply:
Will do!